scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Record.java

```diff
@@ -521,14 +526,14 @@ private Record(ProtocolVersion version, int epoch, long sequenceNumber, long rec
 		plaintext.writeBytes(getBlockCipherMac(writeState, compressedFragment));
 
 		// determine padding length
-		int ciphertextLength = compressedFragment.length + writeState.getCipherSuite().getMacLength() + 1;
-		int smallestMultipleOfBlocksize = writeState.getRecordIvLength();
-		while ( smallestMultipleOfBlocksize <= ciphertextLength) {
-			smallestMultipleOfBlocksize += writeState.getRecordIvLength();
+		int ciphertextLength = compressedFragment.length + writeState.getMacLength() + 1;
```

Comment:
the `+1` is the paddingLenght byte ?

Reply:
yes, I also did a test against to old implementation. It shows, that the old has in some cases extra 32 bytes (caused by the `<=` instead `<`). The added test shows, that it works not too bad :-)